### PR TITLE
add configurable search operator for ListOperation

### PR DESCRIPTION
**Pull request by pxpm**
### BEFORE - What was wrong? What was happening before this PR?

This is an old issue since #1980 . There were workarounds, but we can make developers life easier. A few days a solution to solve this problem pop-up in #4277 but we talked about it and decided that it was better to make it configurable than "hardcode" ilike for some db's and like for others. 

### AFTER - What is happening after this PR?

We still use `like` by default, but the search operator is now configurable, so you can change it globally for ALL CRUDS in the `backpack.operations.list` config file, or per crud `$this->crud->setOperationSetting('searchOperator', 'ilike')` in your `setupListOperation`. 

### How did you achieve that, in technical terms?

Added a config for the operator.

### Is it a breaking change?

No it's not.

### CODE REVIEW

**Review by tabacitu — CHANGES_REQUESTED** [review]
I love it. But let's make it `$searchOperator` not `$seachOperator` 😅

File: src/app/Library/CrudPanel/Traits/Search.php [hunk 1]
```diff
@@ -61,11 +61,13 @@ public function applySearchLogicForColumn($query, $column, $searchTerm)
 
         // sensible fallback search logic, if none was explicitly given
         if ($column['tableColumn']) {
+            $seachOperator = config('backpack.operations.list.searchOperator', 'like');
```
Comment (pxpm) [inline src/app/Library/CrudPanel/Traits/Search.php:64]:
```suggestion
            $searchOperator = config('backpack.operations.list.searchOperator', 'like');
```

File: src/app/Library/CrudPanel/Traits/Search.php [hunk 2]
```diff
@@ -61,11 +61,13 @@ public function applySearchLogicForColumn($query, $column, $searchTerm)
...
+
             switch ($columnType) {
                 case 'email':
                 case 'text':
                 case 'textarea':
-                    $query->orWhere($this->getColumnWithTableNamePrefixed($query, $column['name']), 'like', '%'.$searchTerm.'%');
+                    $query->orWhere($this->getColumnWithTableNamePrefixed($query, $column['name']), $seachOperator, '%'.$searchTerm.'%');
```
Comment (pxpm) [inline src/app/Library/CrudPanel/Traits/Search.php:70]:
```suggestion
                    $query->orWhere($this->getColumnWithTableNamePrefixed($query, $column['name']), $searchOperator, '%'.$searchTerm.'%');
```

File: src/app/Library/CrudPanel/Traits/Search.php [hunk 3]
```diff
@@ -81,8 +83,8 @@ public function applySearchLogicForColumn($query, $column, $searchTerm)
 
                 case 'select':
                 case 'select_multiple':
-                    $query->orWhereHas($column['entity'], function ($q) use ($column, $searchTerm) {
-                        $q->where($this->getColumnWithTableNamePrefixed($q, $column['attribute']), 'like', '%'.$searchTerm.'%');
+                    $query->orWhereHas($column['entity'], function ($q) use ($column, $searchTerm, $seachOperator) {
```
Comment (pxpm) [inline src/app/Library/CrudPanel/Traits/Search.php:86]:
```suggestion
                    $query->orWhereHas($column['entity'], function ($q) use ($column, $searchTerm, $searchOperator) {
```

File: src/app/Library/CrudPanel/Traits/Search.php [hunk 4]
```diff
@@ -81,8 +83,8 @@ public function applySearchLogicForColumn($query, $column, $searchTerm)
...
+                        $q->where($this->getColumnWithTableNamePrefixed($q, $column['attribute']), $seachOperator, '%'.$searchTerm.'%');
```
Comment (pxpm) [inline src/app/Library/CrudPanel/Traits/Search.php:87]:
```suggestion
                        $q->where($this->getColumnWithTableNamePrefixed($q, $column['attribute']), $searchOperator, '%'.$searchTerm.'%');
```

**Comment by tabacitu** [comment]
This. Is. PERFECT!

gutearbeitoriginals-perfect-top-l3q2LH45XElELRzRm

Non-breaking. Good naming. Surgical change. 
Thank you @pxpm ! Well done!

**Comment by siddharth-bhansali** [comment]
I am guessing the relevant change is also needed on the FetchOperation trait to allow for case insensitive search in a select field with an ajax data source.

**Comment by pxpm** [comment]
Hello @siddharth-bhansali and I guess you are right 🙃 

I've submitted the PR to pro that would:

- Make the searchOperator for Fetch by default the same that is defined for List. 
- Added it as a operation configuration, so you can change it if it's different from the List one. 

Thanks for the feedback 🙏

**Comment by pxpm** [comment]
Hello @siddharth-bhansali and I guess you are right 🙃 

I've submitted the PR to pro that would:

- Make the searchOperator for Fetch by default the same that is defined for List. 
- Added it as a operation configuration, so you can change it if it's different from the List one. 

Thanks for the feedback 🙏

**Comment by siddharth-bhansali** [comment]
@pxpm great! I had been following this thread for a while now and eagerly was waiting for the change, however didn't happen with the select. For now I have overridden the trait, but a configurable operator would be great! Thank you very much!

**Pull request merged**